minari/data_collector/data_collector.py

```diff
@@ -192,9 +200,12 @@ def step(
             truncated=truncated,
         )
 
-        # force step data dictionary to include keys corresponding to Gymnasium step returns:
+        # Force step data dictionary to include keys corresponding to Gymnasium step returns:
         # actions, observations, rewards, terminations, truncations, and infos
         assert STEP_DATA_KEYS.issubset(step_data.keys())
+        # Check that the saved observation and action belong to the dataset's observation/action spaces
+        assert self.dataset_observation_space.contains(step_data["observations"])
+        assert self.dataset_action_space.contains(step_data["actions"])
```

Comment:
In general we want to have error messages if the assert fails just for simpler debugging, so assert(x), “x is not true”

Reply:
I added some more descriptive messages to the asserts in `data_collector.py`